### ISSUE TITLE
Missing include..

### ISSUE THREAD

**Issue by martysmind** [ProjectBorealis/UE4GitPlugin#49]
I get a compile error when I try to use the UEGitPlugin with UE5.  It looks like you are missing an include in GitSourceControlMenu.h.  It works once I add this:

#if ENGINE_MAJOR_VERSION == 5
#include "ToolMenuSection.h"
#endif

**Comment by mastercoms** [comment]
What was the compile error you got?

**Comment by martysmind** [comment]
Error dump:

```
Building 6 actions with 6 processes...
[1/6] Resource Default.rc2
[2/6] Compile PCH.GitSourceControl.cpp
[3/6] Compile Module.GitSourceControl.cpp
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.h(42): error C2061: syntax error: identifier 'FToolMenuSection'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(56): error C2660: 'FGitSourceControlMenu::AddMenuExtension': function does not take 1 arguments
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.h(42): note: see declaration of 'FGitSourceControlMenu::AddMenuExtension'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(522): error C2511: 'void FGitSourceControlMenu::AddMenuExtension(FToolMenuSection &)': overloaded member function not found in 'FGitSourceControlMenu'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.h(16): note: see declaration of 'FGitSourceControlMenu'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(535): error C2671: 'FGitSourceControlMenu::AddMenuExtension': static member functions do not have 'this' pointers
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(535): error C2672: 'TDelegate<void (void),FDefaultDelegateUserPolicy>::CreateRaw': no matching overloaded function found
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(536): error C2671: 'FGitSourceControlMenu::AddMenuExtension': static member functions do not have 'this' pointers
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(536): error C2672: 'TDelegate<bool (void),FDefaultDelegateUserPolicy>::CreateRaw': no matching overloaded function found
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(552): error C2671: 'FGitSourceControlMenu::AddMenuExtension': static member functions do not have 'this' pointers
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(552): error C2672: 'TDelegate<void (void),FDefaultDelegateUserPolicy>::CreateRaw': no matching overloaded function found
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(553): error C2671: 'FGitSourceControlMenu::AddMenuExtension': static member functions do not have 'this' pointers
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(553): error C2672: 'TDelegate<bool (void),FDefaultDelegateUserPolicy>::CreateRaw': no matching overloaded function found
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(569): error C2671: 'FGitSourceControlMenu::AddMenuExtension': static member functions do not have 'this' pointers
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(569): error C2672: 'TDelegate<void (void),FDefaultDelegateUserPolicy>::CreateRaw': no matching overloaded function found
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(568): error C2440: '<function-style-cast>': cannot convert from 'FCanExecuteAction' to 'FUIAction'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(571): note: No constructor could take the source type, or constructor overload resolution was ambiguous
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(557): error C2664: 'FToolMenuEntry &FToolMenuSection::AddMenuEntry(const TSharedPtr<const FUICommandInfo,ESPMode::ThreadSafe> &,const TAttribute<FText> &,const TAttribute<FText> &,const TAttribute<FSlateIcon> &,const FName,const TOptional<FName>)': cannot convert argument 1 from 'const char [10]' to 'const TSharedPtr<const FUICommandInfo,ESPMode::ThreadSafe> &'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(572): note: Reason: cannot convert from 'const char [10]' to 'const TSharedPtr<const FUICommandInfo,ESPMode::ThreadSafe>'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(572): note: No constructor could take the source type, or constructor overload resolution was ambiguous
C:\Program Files\Epic Games\UE_5.0\Engine\Source\Developer\ToolMenus\Public\ToolMenuSection.h(38): note: see declaration of 'FToolMenuSection::AddMenuEntry'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(586): error C2671: 'FGitSourceControlMenu::AddMenuExtension': static member functions do not have 'this' pointers
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(586): error C2672: 'TDelegate<void (void),FDefaultDelegateUserPolicy>::CreateRaw': no matching overloaded function found
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(585): error C2440: '<function-style-cast>': cannot convert from 'FCanExecuteAction' to 'FUIAction'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(588): note: No constructor could take the source type, or constructor overload resolution was ambiguous
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(574): error C2664: 'FToolMenuEntry &FToolMenuSection::AddMenuEntry(const TSharedPtr<const FUICommandInfo,ESPMode::ThreadSafe> &,const TAttribute<FText> &,const TAttribute<FText> &,const TAttribute<FSlateIcon> &,const FName,const TOptional<FName>)': cannot convert argument 1 from 'const char [11]' to 'const TSharedPtr<const FUICommandInfo,ESPMode::ThreadSafe> &'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(589): note: Reason: cannot convert from 'const char [11]' to 'const TSharedPtr<const FUICommandInfo,ESPMode::ThreadSafe>'
C:\Users\Marty\source\temprepo2\tempout\HostProject\Plugins\GitSourceControl\Source\GitSourceControl\Private\GitSourceControlMenu.cpp(589): note: No constructor could take the source type, or constructor overload resolution was ambiguous
C:\Program Files\Epic Games\UE_5.0\Engine\Source\Developer\ToolMenus\Public\ToolMenuSection.h(38): note: see declaration of 'FToolMenuSection::AddMenuEntry'
CompilationResultException: Error: OtherCompilationError
   at UnrealBuildTool.ActionGraph.ExecuteActions(BuildConfiguration BuildConfiguration, List`1 ActionsToExecute) in d:\build\++UE5\Sync\Engine\Source\Programs\UnrealBuildTool\System\ActionGraph.cs:line 375
   at UnrealBuildTool.BuildMode.Build(TargetMakefile[] Makefiles, List`1 TargetDescriptors, BuildConfiguration BuildConfiguration, BuildOptions Options, FileReference WriteOutdatedActionsFile) in d:\build\++UE5\Sync\Engine\Source\Programs\UnrealBuildTool\Modes\BuildMode.cs:line 611
   at UnrealBuildTool.BuildMode.Build(List`1 TargetDescriptors, BuildConfiguration BuildConfiguration, ISourceFileWorkingSet WorkingSet, BuildOptions Options, FileReference WriteOutdatedActionsFile, Boolean bSkipPreBuildTargets) in d:\build\++UE5\Sync\Engine\Source\Programs\UnrealBuildTool\Modes\BuildMode.cs:line 282
   at UnrealBuildTool.BuildMode.Execute(CommandLineArguments Arguments) in d:\build\++UE5\Sync\Engine\Source\Programs\UnrealBuildTool\Modes\BuildMode.cs:line 237
   at UnrealBuildTool.UnrealBuildTool.Main(String[] ArgumentsArray) in d:\build\++UE5\Sync\Engine\Source\Programs\UnrealBuildTool\UnrealBuildTool.cs:line 593
WriteFileIfChanged() wrote 8 changed files of 8 requested writes.
Timeline:
```

**Comment by martysmind** [comment]
Another data point, I just tried the HEAD version of SRombauts UE4GitPlugin and it does not have this error.

**Comment by martysmind** [comment]
Updated the description with a specific include to add to GitSourceControlMenu.h.

#if ENGINE_MAJOR_VERSION == 5
#include "ToolMenuSection.h"
#endif

**Comment by mastercoms** [comment]
This was fixed in ef44118a3ac449f5bb752e292a2218b5cfa3767e
